generation/src/proof_generation/pattern.py

```diff
@@ -84,6 +84,14 @@ def evar_is_fresh(self, name: int) -> bool:
     def metavars(self) -> set[int]:
         raise NotImplementedError
 
+    """
+        Returns the set of all free variables occurring in the pattern
+        Makes no guarantees about freshness!
+    """
+
+    def fv(self) -> set[EVar | SVar]:
```

Comment:
Lets rename this to `occuring_vars`. Otherwise it will be mistaken for `free_var` which has a different mathematical definition. It may also easily be mistaken for `fresh_var` so prefer full forms.

```suggestion
    """
        Returns the set of concrete variables occurring free in the pattern.
        Note that this is not the same as the set of free variables as it does
        reason about metavariables.
    """

    def occuring_vars(self) -> set[EVar | SVar]:
```

Reply:
Sounds good.